### DISCUSSION TITLE
Default culture not honored

### DISCUSSION THREAD

**Discussion by DotCat1985** [OrchardCMS/OrchardCore#8440] [Q&A]
Hi,
I customized the login page overridding the view inside the theme as follows:

```
Theme.csproj
 |
 -- Localization
 |   |
 |   -- it.po
 |
 -- Recipes
 |   |
 |   -- Theme.recipe.json
 |
 -- Views
     |
	 -- OrchardCore.Users
	     |
	     -- Account
		     |
	             -- Login.cshtml
```

where:
- `it.po` contains the custom translantion in **Italian** of some overridden views like `Login.cshtml`:

	```
	#: Theme\Views\OrchardCore.Users\Account\Login.cshtml:26
	msgctxt "Theme.Views.OrchardCore.Users.Account.Login"
	msgid "Use a local account to log in"
	msgstr "Accesso all'area riservata"

	#: Theme\Views\OrchardCore.Users\Account\Login.cshtml:30
	msgctxt "Theme.Views.OrchardCore.Users.Account.Login"
	msgid "Username"
	msgstr "Nome utente"

	#: Theme\Views\OrchardCore.Users\Account\Login.cshtml:52
	msgctxt "Theme.Views.OrchardCore.Users.Account.Login"
	msgid "Remember me"
	msgstr "Ricorda la password"

	#: Theme\Views\OrchardCore.Users\Account\Login.cshtml:54
	msgctxt "Theme.Views.OrchardCore.Users.Account.Login"
	msgid "Forgot your password?"
	msgstr "Password dimenticata?"

	#: Theme\Views\OrchardCore.Users\Account\Login.cshtml:57
	msgctxt "Theme.Views.OrchardCore.Users.Account.Login"
	msgid "Log in"
	msgstr "Accedi"
	```

- `Theme.recipe.json` includes the following step about localization setting

 ```
{
      "name": "Settings",
      "LocalizationSettings": {
        "DefaultCulture": "en",
        "SupportedCultures": [ "en", "it" ]
      },
       ...
    }
```

- `it.po` contains **Italian** translantion
- `Login.cshtml` is written as follows:

```
@using Microsoft.AspNetCore.Identity
@using OrchardCore.Entities
@using OrchardCore.Settings
@using OrchardCore.Users
@using OrchardCore.Users.Models
@model OrchardCore.Users.ViewModels.LoginViewModel
@inject SignInManager<IUser> SignInManager
@inject ISiteService SiteService
@inject UserManager<IUser> UserManager;
@{
    ViewLayout = "Layout__Login";
    var allowResetPassword = (await SiteService.GetSiteSettingsAsync()).As<ResetPasswordSettings>().AllowResetPassword;
}
<h6 class="text-center mt-1">@T["Use a local account to log in."]</h6>
<hr />
<form asp-controller="Account" asp-action="Login" asp-route-returnurl="@ViewData["ReturnUrl"]" method="post" class="form-row align-items-center">
    <div asp-validation-summary="All" class="text-danger"></div>
    <div class="form-group col-sm-12">
        <label asp-for="UserName">@T["Username"]</label>
        <input asp-for="UserName" class="form-control" autofocus tabindex="1" required />
        <span asp-validation-for="UserName" class="text-danger"></span>
    </div>
    <div class="form-group col-sm-12">
        <label asp-for="Password">@T["Password"]</label>
        <input asp-for="Password" class="form-control" tabindex="2" required />
        <span asp-validation-for="Password" class="text-danger"></span>
    </div>
    <div class="form-group col-sm-12">
        <div class="custom-control custom-checkbox my-1 mr-sm-2 form-group">
            <input asp-for="RememberMe" class="custom-control-input" tabindex="3">
            <label asp-for="RememberMe" class="custom-control-label">@T["Remember me"]</label>
        </div>
    </div>
    <div class="form-group col-sm-12 text-left mb-3">
        @if (allowResetPassword)
        {
            <a asp-controller="ResetPassword" asp-action="ForgotPassword" class="label-link">@T["Forgot your password?"]</a>
        }
    </div>
    <div class="form-group col-lg-4 col-md-5 col-sm-6">
        <button type="submit" class="btn btn-primary" tabindex="4">@T["Log in"]</button>
    </div>
</form>
```

When I empty cache browser and launch the login page, `@T["Username"]`, `@T["Password"]`, `@T["Remember me"]` and `@T["Forgot your password?"]` returns **Italian** translations instead of **English**. When I change the display language to **English** with **Content Localization Picker** developed with `OrchardCore.ContentLocalization.ContentCulturePicker` module, they corretly return **English**.
How could I fix this issue?
Let me know. Thank you.

**Comment by dodyg** [comment]
If I am not mistaken OC by default get the cue for localization via Accept-Language header. Are you in anyway access this via an Italian OS setting?

**Reply by DotCat1985** [reply]
Maybe yes. Ispecting the source code of OC, it seems that set Italian culture which is the language of my OS.
Should I set the language with cookies or do anything?

**Reply by DotCat1985** [reply]
I've solved the issue removing `Accept-Language` header in `Startup.cs` before calling `AddOrchardCms()` , which include the invocation of `AddMvc()`. Below you find my solution:

```

public class Startup
{
	public void ConfigureServices(IServiceCollection services)
	{
		...

		services.Configure<RequestLocalizationOptions>(options =>
		{
			var requestProvider = options.RequestCultureProviders.OfType<AcceptLanguageHeaderRequestCultureProvider>().First();
			options.RequestCultureProviders.Remove(requestProvider);
		});

		services.AddOrchardCms();

		...
	}
	
	public void Configure(IApplicationBuilder app, IHostEnvironment env)
	{
		...

		var options = app.ApplicationServices.GetService<IOptions<RequestLocalizationOptions>>();
		app.UseRequestLocalization(options.Value);

		app.UseOrchardCore();
		
		...
	}
}

```

**Reply by hishamco** [reply]
> I've solved the issue removing Accept-Language header

What you did is good and bad, good because it solves your issue, but this eliminate the possibility for localization middleware to pickup the selected culture from HTTP header. Why you didn't set the Italian as default culture in the admin cultures

**Reply by Skrypt** [reply]
This is a misunderstanding on how the Localization Middleware works. If your browser has a culture setting set to a culture other than the default one set in Orchard Core. It will display the page in that "secondary" culture because you allow it to. If you don't want to allow people getting the proper culture of your website based on their browser preference. Removing the AcceptLanguageHeaderRequestCultureProvider is the way to go. Else, keeping it should not be an issue either if it's used with a Cookie provider to persist the prefered culture.

So, here it's a false statement to say that the default culture is not honored.

**Reply by alafi** [reply]
> So, here it's a false statement to say that the default culture is not honored.

Your conclusion would be true if the browser allows to send requests without `Accept-Language` header. However, I am using MS Edge and it needs at least one language to be set for the request culture.
So, in 99% of the cases, this language will always be considered by Orchard (in case it is one of the supported cultures of course) and the Default culture option under the localization settings will do nothing.
